### PR TITLE
Update refined to 0.4.0

### PR DESCRIPTION
**Pull request by fthomas**

### CODE REVIEW

**Comment by julien-truffaut** [comment]
:+1:

File: build.sbt [hunk 1]
```diff
@@ -42,7 +42,7 @@ lazy val buildSettings = Seq(
 
 lazy val scalaz     = Def.setting("org.scalaz"      %%% "scalaz-core" % "7.2.1")
 lazy val shapeless  = Def.setting("com.chuusai"     %%% "shapeless"   % "2.3.0")
-lazy val refinedDep = Def.setting("eu.timepit"      %%% "refined"     % "0.3.7")
+lazy val refinedDep = Def.setting("eu.timepit"      %%% "refined"     % "0.4.0")
 lazy val refinedScalaCheckDep = Def.setting("eu.timepit"  %%% "refined-scalacheck" % "0.3.7" % "test")
```
Comment (xuwei-k) [inline build.sbt]: `"refined-scalacheck" % "0.4.0"` ?

Comment (fthomas) [inline build.sbt]: @xuwei-k You're right. Sorry, I messed this up. Will send another PR.

**Comment by julien-truffaut** [comment]
I also missed it, would you mind to reference refined version in `refined-scalacheck` such as we avoid any discrepancy in the future.
